redfish-core/lib/assembly.hpp

```diff
@@ -321,6 +368,80 @@ inline void setAssemblylocationIndicators(
         {
             setLocationIndicatorActive(asyncResp, assembly, iter->second);
         }
+
+        // Handle special case for tod_battery assembly OEM ReadyToRemove property
+        // NOTE: The following method for the special case of the tod_battery
+        // ReadyToRemove property only works when there is only ONE adcsensor
+        // handled by the adcsensor application.
+        if (sdbusplus::message::object_path(assembly).filename() == "tod_battery")
+        {
+            if (!json_util::readJson(req, asyncResp->res, "Assemblies", Assemblies, "Oem", oem))
+            {
+                return;
+            }
+            if (oem)
+            {
+                std::optional<nlohmann::json> openbmc;
+                if (!json_util::readJson(*oem, asyncResp->res, "OpenBMC", openbmc))
+                {
+                    BMCWEB_LOG_ERROR << "Property Missing ";
+                    messages::propertyMissing(asyncResp->res, "OpenBMC");
+                    return;
+                }
+                if (openbmc)
```

Comment:
Can we do a 

    if (!openbmc)
    {
                messages::propertyMissing(asyncResp->res,
                                  pathString + "/OpenBMC");
    }

Reply:
Can you be more specific?  In line 429 I am checking for OpenBMC.

Reply:
if (!json_util::readJson(*oem, asyncResp->res, "OpenBMC", openbmc)) 
                 This doesn't check for OpenBMC.. it just parses it